pkg/service_exporter/service_exporter.go

```diff
@@ -66,52 +68,57 @@ func (e *ServiceQuotasExporter) refreshMetrics() {
 }
 
 func (e *ServiceQuotasExporter) updateMetrics() {
-	quotas, err := e.quotasClient.QuotasAndUsage()
-	if err != nil {
-		log.Fatalf("Could not retrieve quotas and limits: %s", err)
-	}
-
-	for _, quota := range quotas {
-		key := metricKey(quota)
-		log.Infof("Refreshing metrics for resource (%s)", quota.Identifier())
-		if resourceMetric, ok := e.metrics[key]; ok {
-			resourceMetric.usage = quota.Usage
-			resourceMetric.limit = quota.Quota
-			e.metrics[key] = resourceMetric
-		}
-	}
+	e.createQuotasAndDescriptions(true)
 }
 
-func (e *ServiceQuotasExporter) createQuotasAndDescriptions() {
+func (e *ServiceQuotasExporter) createQuotasAndDescriptions(refresh bool) {
 	quotas, err := e.quotasClient.QuotasAndUsage()
 	if err != nil {
 		log.Fatalf("Could not retrieve quotas and limits: %s", err)
 	}
 
 	for _, quota := range quotas {
-		// check so we don't report the same metric more than once
 		key := metricKey(quota)
-		if _, ok := e.metrics[key]; ok {
-			continue
-		}
+		resourceID := quota.Identifier()
 
-		usageHelp := fmt.Sprintf("Used amount of %s", quota.Description)
-		usageDesc := newDesc(e.metricsRegion, quota.Name, "used_total", usageHelp, []string{"resource"})
+		labels := []string{"resource"}
+		labelValues := []string{resourceID}
 
-		limitHelp := fmt.Sprintf("Limit of %s", quota.Description)
-		limitDesc := newDesc(e.metricsRegion, quota.Name, "limit_total", limitHelp, []string{"resource"})
+		for _, tag := range e.includedAWSTags {
+			prometheusFormatTag := servicequotas.ToPrometheusNamingFormat(tag)
+			labels = append(labels, prometheusFormatTag)
+			// Need to set empty label value to keep label name and value count the same
+			labelValues = append(labelValues, quota.Tags[prometheusFormatTag])
+		}
 
-		resourceMetric := Metric{
-			resourceID: quota.Identifier(),
-			usageDesc:  usageDesc,
-			limitDesc:  limitDesc,
-			usage:      quota.Usage,
-			limit:      quota.Quota,
+		if refresh {
```

Comment:
I like the idea of moving that here, but since that adds some additional logic to the `createQuotasAndDescriptions` function, I think it would be great to add a simple test for it :)

Reply:
Added a test to show update doesn't touch descriptions or close the channel, which are things that create would do.

Reply:
Thanks!